homeassistant/components/group/__init__.py

```diff
@@ -341,6 +342,36 @@ async def _async_process_config(hass, config, component):
         )
 
 
+class GroupEntity(Entity):
+    """Representation of a Group of entities."""
+
+    @property
+    def should_poll(self) -> bool:
+        """Disable polling for group."""
+        return False
+
+    async def async_added_to_hass(self):
+        """Register listeners."""
+
+        @callback
+        def _update_at_start(_):
+            self.async_schedule_update_ha_state(True)
+
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, _update_at_start)
+
+    @callback
+    def async_schedule_or_defer_update_ha_state(
+        self, force_refresh: bool = False
+    ) -> None:
+        """Only update once at start."""
+        assert self.hass is not None
+
+        if self.hass.state != CoreState.running:
+            return
+
+        self.async_schedule_update_ha_state(force_refresh)
```

Comment:
There is nothing async about a group, we should be able to trigger the update and `self.async_write_ha_state` without going into async land or scheduling an async function (as this method does)

Reply:
It looked weird to me when I moved it.  I didn't change it because I figured there was something else going on here that I didn't look into.  I'll clean this up.

Reply:
Fixed